### PR TITLE
Hotfix/0.5.1, #13 #14 #15 #17 #18

### PR DESCRIPTION
**Pull request by Craphtex**
Issues fixed (https://trello.com/b/W9o8ISjJ)
#13 Use Meteor.method to generate invoices
#14 Filter buttons do not reflect current state
#15 Stopping subscriptions is not necessary

Non listed issues fixed
- Security: Moved database filter creation to server side
- Maintainability: Removed functionality from view to controller

### CODE REVIEW

File: collections/DBFilter.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Meteor.methods({
+  getFilter: function(date, days) {
```
Comment (DominikGuzei) [inline collections/DBFilter.js]: So you are using a method to retrieve the DB filter? Can you explain what the benefit of this will be?

**Comment by DominikGuzei** [comment]
Im not able to invoke the Meteor.method from the UI now?

**Comment by DominikGuzei** [comment]
Maybe just move the invoice generation to a `Meteor.startup` callback on the server side that checks if there are any existing invoices. It was not a requirement that i can generate invoices via the UI

**Comment by DominikGuzei** [comment]
im getting this error on the client console:

```
Exception in template helper: ReferenceError: filter is not defined
    at Object.Template.InvoiceTable.helpers.invoices (http://localhost:3000/app/client/InvoiceTable.js?fe077330337e61c755577c43ad1303fd21a7d375:11:42)
    at http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:2986:16
    at http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:1650:16
    at http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:3038:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:3671:12)
    at http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:3037:27
    at Object.Spacebars.call (http://localhost:3000/packages/spacebars.js?1aedcc2aa3ae9ff5d860d73516110cedd77c033e:167:18)
    at http://localhost:3000/app/client/template.InvoiceTable.js?7377749cbcbc57b0e49139720f8b508696604b15:14:22
    at .<anonymous> (http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:2751:17)
    at http://localhost:3000/packages/blaze.js?9391df93ba5076c2cfc61ee68724eb79b65f00d9:1872:20
```

File: collections/DBFilter.js (hunk 1 above)
Comment (Craphtex) [inline collections/DBFilter.js]: The description given in the tutorials on the meteor home page when making a call is:
1. The client sends a request to the server to run the method in a secure environment, just like an AJAX request would work
2. A simulation of the method runs directly on the client to attempt to predict the outcome of the server call using the available information

For me it seemed like a clean way to make the same filter available for both the server and the client.

Comment (DominikGuzei) [inline collections/DBFilter.js]: You only need to make method calls if the client needs information from the server that it cannot calculate / obtain (eg from the DB) itself. The filtering logic is just some lines of code that can shared between client and server without having to call a method :wink:

Take a look at the code of the others, how they solved the problem by adding a static `InvoicesCollection.byTimeRange` method which takes 3 params and returns a query cursor.

**Comment by Craphtex** [comment]
The error seems to stem from incorrect arguments when calling getFilter.
This is definitely a bug, and it appears when entering via `/`.
There seems to be a problem when the BlazeLayout is passing an empty object as well, it will then spappear as `undefined`.

**Comment by DominikGuzei** [comment]
Seems like another good reason to remove the `getFilter` method :stuck_out_tongue:

**Comment by DominikGuzei** [comment]
@Craphtex are you going to fix the mentioned problems? I'd like to see a fully working example tomorrow :wink:

**Comment by Craphtex** [comment]
Should be working as intended now :)

**Comment by DominikGuzei** [comment]
Ok it works now – the only thing missing is that you refactor the `generateInvoices` method into a simple `Meteor.startup` on the server, where you just do it, without having to explicitly invoke that method! Will merge now anyway so that you can continue in proper feature / fix branches :wink:

**Pull request merged**